src/lib/y2partitioner/sequences/transaction_wizard.rb

```diff
@@ -0,0 +1,61 @@
+require "yast"
+require "ui/sequence"
+require "y2partitioner/device_graphs"
+
+Yast.import "Wizard"
+
+module Y2Partitioner
+  module Sequences
+    # Base class for the sequences that modify the devicegraph with a
+    # transaction (most of the expert partitioner sequences)
+    class TransactionWizard < UI::Sequence
+      include Yast::Logger
+
+      def initialize
+        textdomain "storage"
+      end
+
+      # Main method of any UI::Sequence.
+      #
+      # See the associated documentation at yast2.
+      def run
+        sym = nil
+        DeviceGraphs.instance.transaction do
+          init_transaction
+
+          sym = wizard_next_back do
+            super(sequence: sequence_hash)
+          end
+
+          sym == :finish
+        end
+        sym
+      end
+
+    protected
+
+      # Specification of the steps of the sequence.
+      #
+      # To be defined by each subclass.
+      #
+      # See UI::Sequence in yast2.
+      def sequence_hash
```

Comment:
Maybe to define this as an abstract method?

Reply:
Hmm, maybe. I was also unsure. But as an exception, I will merge as it is... because it's my last day before a long absence and I don't want to go through all the testing cycle again. :wink: